src/views/ProtestPage.js

```diff
@@ -17,11 +17,37 @@ import {
 } from 'react-share';
 import SocialButton, { Button } from '../components/Button/SocialButton';
 import * as texts from './ProtestPageTexts.json';
-import { dateToDayOfWeek, formatDate, isAdmin, sortDateTimeList } from '../utils';
+import { dateToDayOfWeek, formatDate, isAdmin, sortDateTimeList, isAuthenticated, isVisitor} from '../utils';
 import ProtectedRoute from '../components/ProtectedRoute/ProtectedRoute';
 
+
 const mobile = `@media (max-width: 500px)`;
 
+function getEditButtonLink(user, protest) {
+  const editRoute = `/protest/${protest.id}/edit`;
+  
+  if (isAdmin(user)) {
+    return editRoute;
+  }
+
+  if (isVisitor(user)) {
+    // Sign up before redirected to leader request
+    return `/sign-up?returnUrl=/leader-request?protest=${protest.id}`;
+  }
+
+  if (isAuthenticated(user)) {
+    // The user is a leader
+    if (protest?.roles?.leader?.includes(user.uid)) {
```

Comment:
@ranyitz `isLeader` logic should also be extracted to `utils`

Reply:
Agree